tokenizers/src/pre_tokenizers/split.rs

```diff
@@ -1,145 +1,231 @@
-                    })
-                }  
-            },
-            MergedWithNext => {
-                if self.invert { 
-                    pretokenized.split(|_, normalized| {
-                        normalized.split(Invert(&self.re), MergedWithNext)
-                    })
-                } else {
-                    pretokenized.split(|_, normalized| {
-                        normalized.split(&self.re, MergedWithNext)
-                    })
-                }  
-            },
-            Contiguous => {
-                if self.invert { 
-                    pretokenized.split(|_, normalized| {
-                        normalized.split(Invert(&self.re), Contiguous)
-                    })
-                } else {
-                    pretokenized.split(|_, normalized| {
-                        normalized.split(&self.re, Contiguous)
-                    })
-                }  
-            },
+        if self.invert { 
+            pretokenized.split(|_, normalized| {
+                normalized.split(Invert(&self.regex), self.behavior)
+            })
+        } else {
+            pretokenized.split(|_, normalized| {
+                normalized.split(&self.regex, self.behavior)
+            })
         }
     }
 }
 
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use SplitDelimiterBehavior::*;
+    use crate::{OffsetReferential, OffsetType, PreTokenizer};
+
+    #[test]
+    fn basic() {
+        let tests = vec![
+            (
+                Removed,
+                "How are you doing?",
+                vec![
+                    ("How", (0, 3)),
+                    ("are", (4, 7)),
+                    ("you", (8, 11)),
+                    ("doing", (12, 17)),
+                    ("?", (17, 18)),
+                ],
+            ),
+            (
+                Isolated,
+                "How are you doing?",
+                vec![
+                    ("How", (0, 3)),
+                    (" ", (3, 4)),
+                    ("are", (4, 7)),
+                    (" ", (7, 8)),
+                    ("you", (8, 11)),
+                    (" ", (11, 12)),
+                    ("doing", (12, 17)),
+                    ("?", (17, 18)),
+                ],
+            ),
+            (
+                MergedWithPrevious,
+                "How are you doing?",
+                vec![
+                    ("How ", (0, 4)),
+                    ("are ", (4, 8)),
+                    ("you ", (8, 12)),
+                    ("doing", (12, 17)),
+                    ("?", (17, 18)),
+                ],
+            ),
+            (
+                MergedWithNext,
+                "How are you doing?",
+                vec![
+                    ("How", (0, 3)),
+                    (" are", (3, 7)),
+                    (" you", (7, 11)),
+                    (" doing", (11, 17)),
+                    ("?", (17, 18)),
+                ],
+            ),
+            (
+                Contiguous,
```

Comment:
Just to make sure that Isolated and Contiguous are different because:

Isolated : "I  have this"  -> "I", " ", " ", "have", " ", "this"
Contiguous : "I  have this"  -> "I", "  ", "have", " ", "this"
(Not sure it's how it's going to behave that way, and even if it's the correct behavior with this regexp, but fyi)